torch/_dynamo/variables/tensor.py

```diff
@@ -333,6 +333,23 @@ def method_attr__version(self, tx):
             tx, [self], {}
         )
 
+    def call_hasattr(self, tx, name):
+        try:
+            self.var_getattr(tx, name)
+            ret_val = True
+        except (NotImplementedError, AttributeError):
+            ret_val = False
+
+        if self.source:
+            install_guard(
+                AttrSource(self.source, name).make_guard(GuardBuilder.HASATTR)
+            )
+            ret_val = ret_val or name in self.proxy.node.meta.get("tensor_dict", dict())
```

Comment:
Maybe this should go in `var_getattr`

```
self.proxy.node.meta.get("tensor_dict", dict())
```

Reply:
oh sure I can add this in there

Reply:
Actually I'm going to remove, this is equivalent to evaluating the input source and checking its attributes which we already do.